accounts/views/connection_views.py

```diff
@@ -64,3 +65,16 @@ def connections(request):
         return create_connection(request)
     elif request.method == "GET":
         return retrieve_connections(request)
+
+
+@api_view(["PATCH", "GET"])
+@permission_classes(
+    [
+        IsAuthenticated,
+    ]
+)
+def connection(request):
```

Comment:
- you forget to get the `connection_id` as a parameter for this view function.
- also, you forgot to pass it to the called functions(`update_connection`, `retrieve_single_connection`).